platform/mellanox/mlnx-platform-api/sonic_platform/chassis.py

```diff
@@ -95,5 +129,169 @@ def _get_port_position_tuple_by_sku_name(self):
         position_tuple = port_position_tuple_list[hwsku_dict[out.rstrip('\n')]]
         return position_tuple
 
-    
+    def get_base_mac(self):
+        """
+        Retrieves the base MAC address for the chassis
+
+        Returns:
+            A string containing the MAC address in the format
+            'XX:XX:XX:XX:XX:XX'
+        """
+        return self.eeprom.get_base_mac()
+
+    def get_serial_number(self):
+        """
+        Retrieves the hardware serial number for the chassis
+
+        Returns:
+            A string containing the hardware serial number for this chassis.
+        """
+        return self.eeprom.get_serial_number()
+
+    def get_system_eeprom_info(self):
+        """
+        Retrieves the full content of system EEPROM information for the chassis
+
+        Returns:
+            A dictionary where keys are the type code defined in
+            OCP ONIE TlvInfo EEPROM format and values are their corresponding
+            values.
+        """
+        return self.eeprom.get_system_eeprom_info()
+
+    def _read_generic_file(self, filename, len):
+        """
+        Read a generic file, returns the contents of the file
+        """
+        r = ''
+        try:
+            f = io.open(filename)
+            r = f.read(len)
+            f.close()
+            return r
+        except:
+            return ''
+
+    def _verify_reboot_cause(self, filename):
+        '''
+        Open and read the reboot cause file in 
+        /var/run/hwmanagement/system (which is defined as REBOOT_CAUSE_ROOT)
+        If a reboot cause file doesn't exists, returns '0'.
```

Comment:
should u consider not existing file as a WARNING/ERROR? might be good to have such note in the LOG.
need to consider if this is an error and if so 0 should not be returned,

Reply:
Not all reboot cause files exist in all SKUs. For example, reset_hotswap_or_wd which indicates whether the system is rebooted by watchdog, exists in 2410 but not in 3700.
I'm not sure whether it is a normal state or an issue. Only if it is an issue should we report a log.
I think 0 should be returned when the file doesn't exist since returning 1 means it caused last time reboot but obvious it's not true.

Reply:
Having discussed with Kebo, I decide to add a log when the file doesn't exist.
But it still returns 0 in that case.